extensions/arc/deployment/src/test/java/io/quarkus/arc/test/cp/ContextProviderDisabledTest.java

```diff
@@ -0,0 +1,74 @@
+package io.quarkus.arc.test.cp;
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.junit.jupiter.api.Assertions.assertFalse;
+import static org.junit.jupiter.api.Assertions.fail;
+
+import java.util.concurrent.ExecutionException;
+import java.util.concurrent.TimeUnit;
+import java.util.concurrent.TimeoutException;
+
+import javax.annotation.PostConstruct;
+import javax.enterprise.context.ContextNotActiveException;
+import javax.enterprise.context.RequestScoped;
+import javax.inject.Inject;
+
+import org.eclipse.microprofile.context.ManagedExecutor;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.extension.RegisterExtension;
+
+import io.quarkus.arc.Arc;
+import io.quarkus.arc.ManagedContext;
+import io.quarkus.test.QuarkusUnitTest;
+
+public class ContextProviderDisabledTest {
+
+    @RegisterExtension
+    static final QuarkusUnitTest config = new QuarkusUnitTest().overrideConfigKey("quarkus.arc.context-propagation.enabled",
+            "false");
+
+    @Inject
+    ManagedExecutor all;
+
+    @Inject
+    MyRequestBean bean;
+
+    @Test
+    public void testPropagationDisabled() throws InterruptedException, ExecutionException, TimeoutException {
+        ManagedContext requestContext = Arc.container().requestContext();
+
+        requestContext.activate();
+        assertEquals("FOO", bean.getId());
+        try {
+            assertEquals("OK",
+                    all.completedFuture("OK").thenApplyAsync(text -> {
+                        assertFalse(requestContext.isActive());
+                        try {
+                            bean.getId();
+                            fail();
```

Comment:
IIUC, this would throw on a different thread, no? If so, that wouldn't really fail the test -- at least not immediately. The test would fail because the future wouldn't complete in time. To make the intent more obvious, I'd remove this `fail()` call and instead, in the `ContextNotActiveException` handler, set an `AtomicBoolean` to `true`. And at the end of this test method, assert that the `AtomicBoolean` is indeed `true`.

Reply:
+1, I was just about to write a very similar comment :)

Reply:
Actually there's more than just this `fail()` call, there's also an assertion a few lines above. Again, if I'm reading this correctly, the entire thing will run on a different thread, so if that assert failed, it wouldn't fail the test directly. (Again, the test would fail eventually, but...)

Or is this intentional?

Reply:
It's intentional. If an unchecked exception is thrown inside a "computation" then all dependent stages complete exceptionally and thus `toCompletableFuture().get(5, TimeUnit.SECONDS)` would result in an `ExecutionException` (wrapping an `CompletionException` wrapping the original exception). We don't care about threads at all...

Reply:
Fair enough. I'd at least add a comment, but that may be just me :-)

Reply:
A good comment is always useful but in this case I thought that it's kind of obvious because this is how `CompletionStage`/`CompletableFuture` works.